security-policies/bundle/compliance/cis_azure/rules/cis_3_11/rule.rego

```diff
@@ -0,0 +1,29 @@
+package compliance.cis_azure.rules.cis_3_11
+
+import data.compliance.lib.common
+import data.compliance.policy.azure.data_adapter
+import future.keywords.if
+
+finding = result if {
+	# filter
+	data_adapter.is_storage_account
+	_ = data_adapter.resource.extension.blobService
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(soft_delete_is_enabled),
+		{"Resource": data_adapter.resource},
+	)
+}
+
+default soft_delete_is_enabled = false
+
+soft_delete_is_enabled if {
+	is_policy_valid(data_adapter.resource.extension.blobService.properties.deleteRetentionPolicy)
+	is_policy_valid(data_adapter.resource.extension.blobService.properties.containerDeleteRetentionPolicy)
+}
+
+is_policy_valid(policy) if {
+	policy.enabled == true
+	policy.days > 0
```

Comment:
Just double checking, are you 100% sure this will come as integer?  No possibility of days being a string?

Reply:
Yes, it's int in the response object in SDK. ([DeleteRetentionPolicy](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage@v1.5.0#DeleteRetentionPolicy))